crates/cargo-contract/src/cmd/extrinsics/balance.rs

```diff
@@ -0,0 +1,289 @@
+            .as_str()
+            .context("error converting symbol to string")?;
+        let denomination: u128 = format!("1{}", "0".repeat(decimals as usize)).parse()?;
+        Ok(Self {
+            denomination,
+            symbol: symbol.to_string(),
+        })
+    }
+}
+
+impl BalanceVariant {
+    /// Converts BalanceVariant into Balance.
+    ///
+    /// Throws Error if [BalanceVariant::Denominated(String)] is in an incorrect format.
+    pub fn denominate_balance(&self, token_metadata: &TokenMetadata) -> Result<Balance> {
+        match self {
+            BalanceVariant::Default(balance) => Ok(*balance),
+            BalanceVariant::Denominated(input) => {
+                if let Some(balance_str) =
+                    input.strip_suffix(&format!("k{}", token_metadata.symbol))
+                {
+                    let balance = Decimal::from_str(balance_str)?;
+                    Ok(<Balance>::try_from(balance * dec!(1_000))?
+                        * token_metadata.denomination)
+                } else if let Some(balance_str) =
+                    input.strip_suffix(&format!("M{}", token_metadata.symbol))
+                {
+                    let balance = Decimal::from_str(balance_str)?;
+                    Ok(<Balance>::try_from(balance * dec!(1_000_000))?
+                        * token_metadata.denomination)
+                } else if let Some(balance_str) =
+                    input.strip_suffix(&format!("n{}", token_metadata.symbol))
+                {
+                    let balance = Decimal::from_str(balance_str)?;
+                    Ok(<Balance>::try_from(balance * dec!(1_000_000_000))?
+                        * token_metadata.denomination)
+                } else if let Some(balance_str) =
+                    input.strip_suffix(&format!("μ{}", token_metadata.symbol))
+                {
+                    let balance = Decimal::from_str(balance_str)?;
+                    Ok(<Balance>::try_from(balance * dec!(1_000_000))?)
+                } else if let Some(balance_str) =
+                    input.strip_suffix(&format!("m{}", token_metadata.symbol))
+                {
+                    let balance = Decimal::from_str(balance_str)?;
+                    Ok(<Balance>::try_from(balance * dec!(1_000))?)
+                } else if let Some(balance_str) =
+                    input.strip_suffix(&token_metadata.symbol)
+                {
+                    let balance = Decimal::from_str(balance_str)?;
+                    Ok(<Balance>::try_from(
+                        balance * Decimal::from(token_metadata.denomination),
+                    )?)
+                } else {
+                    let balance = Decimal::from_str(input)?;
+                    Ok(<Balance>::try_from(
+                        balance * Decimal::from(token_metadata.denomination),
+                    )?)
+                }
+            }
+        }
+    }
+
+    /// Display token units in a denominated format.
```

Comment:
Please add some docs for how this method should behave, perhaps with some doctest examples.

Also we need some comprehensive unit tests for this, there aren't any at the moment. These are important because this method is controlling the value displayed to the user which *must* be accurate and with no loss of precision.